scripts/root_dir.mjs

```diff
@@ -0,0 +1,32 @@
+// Copyright 2022 The Outline Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import url from "url";
+
+const FILE_PREFIX = "file://";
+
+// WARNING: if you move this file, you MUST update this file path
+const PATH_FROM_ROOT_TO_THIS_FILE = "/scripts/root_dir.mjs";
+
+export function rootDir() {
+  return import.meta.url.replace(FILE_PREFIX, "").replace(PATH_FROM_ROOT_TO_THIS_FILE, "");
```

Comment:
No string manipulation needed for paths.

Use `URL.pathname instead`:
![image](https://user-images.githubusercontent.com/113565/160437125-a2256352-661e-4e32-8f89-17a36a20f0d4.png)

And `path.resolve(script_path, '..')`

Reply:
good idea

Reply:
yikes it seems... `path.resolve` is broken? https://github.com/Jigsaw-Code/outline-client/runs/5728422776?check_suite_focus=true#step:5:10

Reply:
It's because `pathname` leaves a leading slash where the windows root doesn't actually have one, so path assumes it needs the root drive prefix again.

Reply:
Personally I don't recommend to manipulate urls or paths manually because it is very different across platforms. I would rather use some built-in functions like `dirname` or `fileURLToPath`.